### ISSUE TITLE
[Storage] AWS S3 Bucket Share between two specific users

### ISSUE THREAD

**Issue by OrangeDrangon** [aws-amplify/amplify-js#2694]
** Which Category is your question related to? **
Storage
** What AWS Services are you utilizing? **
Auth and Storage
** Provide additional details e.g. code snippets **
I am building an app where I would like to send a picture to another authenticated user that is only visible to the user that receives it (if the sender can see it that is fine).  The `protected` level looks promising but as far as I understand allows for anyone that is authenticated. I was wondering if this is even possible using amplify or if I am misunderstanding something.

**Comment by manueliglesias** [comment]
Hi @OrangeDrangon 

I think it is not supported, the levels are configured by the Amplify CLI and are these: [File Access Levels](https://aws-amplify.github.io/docs/js/storage#file-access-levels) and as you pointed out, `protected` means readable by all users.

> Protected: Readable by all users, but writable only by the creating user. Files are stored under protected/{user_identity_id}/ where the user_identity_id corresponds to the unique Amazon Cognito Identity ID for that user.

I think I understand your use case, I'll be labeling this issue as a feature request.

**Comment by OrangeDrangon** [comment]
To further flush out what I was saying

User One -> Image provider
User Two -> Image consumer
User Three -> Crafty User

`User one` signs in and has a picture they want to send to `User Two`. The flow I want is `User One` to upload to s3 and restrict access to either `User Two` or `User One + User Two`. This means that `User Three` would have no way to see the image sent between the two other users.

**Comment by OrangeDrangon** [comment]
As I am relatively new to the AWS ecosystem could this be accomplished with a Lambda function of some nature that has admin permission in the S3 ecosystem.

**Comment by manueliglesias** [comment]
> could this be accomplished with a Lambda function of some nature that has admin permission in the S3 ecosystem.

Yes, this is possible. I would start by looking into how S3 [manages permissions](https://docs.aws.amazon.com/AmazonS3/latest/dev/s3-access-control.html).

Maybe this SO answer is also a good start https://stackoverflow.com/questions/7687841/is-it-possible-to-share-files-for-selective-users-using-amazon-s3

**Comment by OrangeDrangon** [comment]
Thanks I will look into that for now. Thanks. I will leave this open in case you guys want to pursue this.
